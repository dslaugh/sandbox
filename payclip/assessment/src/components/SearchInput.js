import React from "react";

export default function SearchInput(props) {
	return <input type="text" id="search" onKeyUp={ props.onKeyUp } placeholder="Search..." />;
}
