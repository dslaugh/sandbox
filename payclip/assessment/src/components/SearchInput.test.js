import React from 'react';
import renderer from 'react-test-renderer';
import SearchInput from './SearchInput';

const mockOnKeyUp = jest.fn();

describe('<SearchInput />', () => {
	it('should render correctly', () => {
		const component = renderer.create(
			<SearchInput onKeyUp={ mockOnKeyUp }/>
		);
		let tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should call the event handler', () => {
		const component = renderer.create(
			<SearchInput onKeyUp={ mockOnKeyUp }/>
		);
		let tree = component.toJSON();

		const e = { target: { value: '1.00' } };
		tree.props.onKeyUp(e);
		tree = component.toJSON();

		expect(mockOnKeyUp).toHaveBeenCalledWith({ target: { value: '1.00' } });
	});
});