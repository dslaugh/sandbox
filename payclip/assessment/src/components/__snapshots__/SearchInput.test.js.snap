// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render correctly 1`] = `
<input
  id="search"
  onKeyUp={[Function]}
  placeholder="Search..."
  type="text"
/>
`;
